### PR TITLE
Shipa 1872

### PR DESCRIPTION
**Pull request by DavisFrench**
# Description

Extend app crd functionality with resources

Fixes # 1872

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Chore (documentation addition or typo, file relocation)

## Testing

- [x] New tests were added with this PR that prove my fix is effective or that my feature works (describe below this bullet)
- [ ] This change requires no testing (i.e. documentation update)

## Documentation

- [x] All added public packages, funcs, and types have been documented with doc comments
- [x] I have commented my code, particularly in hard-to-understand areas

## Final Checklist:

- [x] I followed standard [GitHub flow](https://guides.github.com/introduction/flow/) guidelines
- [x] I have performed a self-review of my own code
- [ ] My changes generate no new warnings

## Additional Information (omit if empty)

Please include anything else relevant to this PR that may be useful to know.

### CODE REVIEW

File: internal/api/v1beta1/app_types.go [hunk 1]
```diff
@@ -90,17 +92,17 @@ func (v DeploymentVersion) String() string {
...
-	RoutingSettings RoutingSettings   `json:"routingSettings,omitempty"`
-	ExposedPorts    []ExposedPort     `json:"exposedPorts,omitempty"`
-
 	// ImagePullSecrets contains a list of secrets to pull the image of this deployment.
 	// If this list is defined, app.Spec.DockerRegistrySpec is not used.
 	ImagePullSecrets []v1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
+	Image            string                    `json:"image"`
+	Version          DeploymentVersion         `json:"version"`
+	Processes        []ProcessSpec             `json:"processes,omitempty"`
+	Resources        v1.ResourceRequirements   `json:"resources,omitempty"`
```
Comment (aleksej-paschenko) [inline internal/api/v1beta1/app_types.go:101]: I think we are good with having resources in ProcessSpec only.
Is this field used?

**Review by aleksej-paschenko — CHANGES_REQUESTED** [review]

File: internal/api/v1beta1/app_types.go (hunk 1 above)
Comment (DavisFrench) [reply internal/api/v1beta1/app_types.go:101]: Good catch, I think I accidentally left this in when I was rebasing. I'll remove it real quick.

**Review by aleksej-paschenko — APPROVED** [review]
looks awesome!

**Review by stinkyfingers — APPROVED** [review]
LGTM

**Pull request merged**